src/Widgets/ReleaseRow.vala

```diff
@@ -0,0 +1,93 @@
+/*-
+ * Copyright (c) 2017 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Adam Bieńkowski <donadigos159@gmail.com>
+ */
+
+public class AppCenter.Widgets.ReleaseRow : Gtk.ListBoxRow {
+    public AppStream.Release release { get; construct; }
+
+    private Gtk.Label header_label;
+    private Gtk.Label description_label;
+
+    construct {
+        string header = format_release_header (release);
+        string description = format_release_description (release);
+
+        header_label = new Gtk.Label (header);
+        header_label.halign = Gtk.Align.START;
+        header_label.get_style_context ().add_class ("h3");
+        header_label.use_markup = true;
+
+        description_label = new Gtk.Label (description);
+        description_label.selectable = true;
+        description_label.get_style_context ().add_class ("h3");
+        description_label.halign = Gtk.Align.START;
+        description_label.margin_start = 12;
+        description_label.use_markup = true;
+
+        var grid = new Gtk.Grid ();
+        grid.margin_bottom = 6;
+        grid.attach (header_label, 0, 0, 1, 1);
+        grid.attach (description_label, 0, 1, 1, 1);
+
+        add (grid);
+    }
+
+    public ReleaseRow (AppStream.Release release) {
+        Object (release: release);
+    }
+
+    private static string format_release_header (AppStream.Release release) {
+        string label;
+
+        unowned string version = release.get_version ();
+
+        uint64 timestamp = release.get_timestamp ();
+        if (timestamp != 0) {
```

Comment:
I attempted to check if the `version != null`, but here we want to check for the timestamp first so we can later define if we do not have the version and only have the date show up.